src/helper_functions.py

```diff
@@ -0,0 +1,99 @@
+# GeoFin : a python library for portfolio optimization and index replication
+# GeoFin is part of GeomScale project
+
+# Copyright (c) 2024 Cyril Bachelard
+# Copyright (c) 2024 Minh Ha Ho
+
+# Licensed under GNU LGPL.3, see LICENCE file
+
+
+
+from typing import Dict
+import numpy as np
+import pandas as pd
+import pickle
+import matplotlib.pyplot as plt
+
+def nearestPD(A):
+    """Find the nearest positive-definite matrix to input
+
+    A Python/Numpy port of John D'Errico's `nearestSPD` MATLAB code [1], which
+    credits [2].
+
+    [1] https://www.mathworks.com/matlabcentral/fileexchange/42885-nearestspd
```

Comment:
Is this code translation from matlab [1] written by you? If yes, please write it in the function description here.

Reply:
it's written by Cyril. I added it in the comment